### PR TITLE
feature(actions): remove php execution timeout for all actions #7204

### LINKED ISSUE #7204: Remove timeout from all actions
Especially delete/disable/enable activities can result in a long run time. Removing the limit for all actions seems the simplest way to avoid ever breaking one of these processes.

We should also log a warning when a script runs long. How long being configurable...

### PR DESCRIPTION
**Pull request by Phanoix**
This removes the php execution timeout for all registered actions to fix #7204.

### CODE REVIEW

**Comment by hypeJunction** [comment]
I'd prefer a reasonably high limit rather than no limit. If you are doing something that runs for 9 million seconds you should probably not be doing that in an action. There are bound to be memory issues as well unless the code doesn't leak any memory for the duration of the action.

**Comment by Phanoix** [comment]
That's a good point, maybe the limit should be added as a setting?

**Comment by Phanoix** [comment]
so something like:
``` php
if ( ini_get('max_execution_time') < 120 );
    set_time_limit(120);
```

**Comment by hypeJunction** [comment]
Playing with ini values is asking for trouble, IMO. I bet there are compilations that use different values/vars, perhaps xdebug chimes in as well.
Let's maybe add an Elgg $CONFIG parameter with no default, let site admins configure if they experience time outs?

**Comment by Phanoix** [comment]
Updated, it will now check $CONFIG->action_time_limit and set the timeout to that if it is set.

File: engine/classes/Elgg/ActionsService.php [hunk 1]
```diff
@@ -173,6 +175,11 @@ public function execute($action, $forwarder = "") {
 			return $forward('actionnotfound', ELGG_HTTP_NOT_IMPLEMENTED);
 		}
 
+		// set the maximum execution time for actions
+		if (isset($CONFIG->action_time_limit)) {
```
Comment (jdalsem) [inline engine/classes/Elgg/ActionsService.php]: can we use elgg_get_config?

**Comment by jdalsem** [comment]
as this is actually a feature and not a bugfix, this should go against the master branch... also update the commit message to be a feature...

**Review by hypeJunction — CHANGES_REQUESTED** [review]
Could you also add it to setting example with a suggested value of 120? and some comments. That way it will get copied over on new installations

**Comment by hypeJunction** [comment]
@Phanoix In my comment, I was referring to https://github.com/Elgg/Elgg/blob/master/elgg-config/settings.example.php

File: engine/classes/Elgg/ActionsService.php (hunk 1 above)
Comment (jeabakker) [inline engine/classes/Elgg/ActionsService.php]: or better yet `$this->config->get();`

**Comment by Phanoix** [comment]
It's now getting the setting from $this->config->get('action_time_limit'); and I added a section for it in the example settings.

**Comment by jdalsem** [comment]
We are almost there... the PR should be on master instead of 2.3 as 2.3 only accepts bugfixes and this is a feature. Otherwise this LGTM

Oh and can you add something to the docs? in docs/info/config.php?

**Comment by Phanoix** [comment]
Changed to merge into master, I'll replicate the changes off the master branch.

**Comment by mrclay** [comment]
@Phanoix thanks! Sorry this sat so long.

**Pull request merged**